### PR TITLE
Fix /eth/v1/node/syncing

### PR DESCRIPTION
**Pull request by cheatfate**
Fix REST server `/eth/v1/node/syncing` call to return proper values even if SyncManager is not running.

### CODE REVIEW

File: beacon_chain/rpc/rest_node_api.nim [hunk 1]
```diff
@@ -254,7 +253,15 @@ proc installNodeApiHandlers*(router: var RestRouter, node: BeaconNode) =
...
   # https://ethereum.github.io/beacon-APIs/#/Node/getSyncingStatus
   router.api(MethodGet, "/eth/v1/node/syncing") do () -> RestApiResponse:
-    return RestApiResponse.jsonResponse(node.syncManager.getInfo())
+    let
+      wallSlot = node.beaconClock.now().slotOrZero()
+      headSlot = node.dag.head.slot
+      distance = wallSlot - headSlot
+      info = RestSyncInfo(
+        head_slot: headSlot, sync_distance: distance,
+        is_syncing: distance != 0'u64
```
Comment (arnetheduck) [inline beacon_chain/rpc/rest_node_api.nim]: this is too strict, ie it'll show up as syncing at the start of every slot, in the period between the start of the slot and until we've received a block. why is syncmanager itself not giving a correct value here?

Comment (cheatfate) [reply beacon_chain/rpc/rest_node_api.nim]: Because there so many different type of sync, that syncmanager is not always running.

Comment (arnetheduck) [reply beacon_chain/rpc/rest_node_api.nim]: This call should be reporting the progress of the ordinary forward sync that we use to chase head after the node goes offline

Comment (cheatfate) [reply beacon_chain/rpc/rest_node_api.nim]: > Requests the beacon node to describe if it's currently syncing or not, and if it is, what block it is up to.

There is no mention on type of sync.
Also people are using this function to check progress of checkpoint sync.

Comment (zah) [reply beacon_chain/rpc/rest_node_api.nim]: I also think the expected definition of "synced" is "able to attest for the current head" (in other words, how far is our current head from the wall clock head). For the other types of syncing, we probably need separate API calls.

**Review by etan-status — APPROVED** [review]

**Pull request merged**